src/io/flutter/jxbrowser/JxBrowserManager.java

```diff
@@ -0,0 +1,227 @@
+// the class path.
+public class JxBrowserManager {
+  private static JxBrowserManager manager;
+  private static final String DOWNLOAD_PATH = FileUtils.platformPath();
+  private static final AtomicReference<JxBrowserStatus> status = new AtomicReference<>(JxBrowserStatus.NOT_INSTALLED);
+  private static final Logger LOG = Logger.getInstance(JxBrowserManager.class);
+  // We will be gating JxBrowser features until all of the features are landed.
+  // To test JxBrowser, set this to true and also add license key to VM options (-Djxbrowser.license.key=<key>).
+  public static final boolean ENABLE_JX_BROWSER = false;
+
+  private JxBrowserManager() {}
+
+  public static JxBrowserManager getInstance() {
+    if (manager == null) {
+      return new JxBrowserManager();
+    }
+    return manager;
+  }
+
+  public boolean isInstalled() {
+    return status.get().equals(JxBrowserStatus.INSTALLED);
+  }
+
+  public void setUp(Project project) {
+    if (!status.compareAndSet(JxBrowserStatus.NOT_INSTALLED, JxBrowserStatus.INSTALLATION_IN_PROGRESS)) {
+      // This check ensures that an IDE only downloads and installs JxBrowser once, even if multiple projects are open.
+      // If already in progress, let calling point wait until success or failure (it may make sense to call setUp but proceed).
+      // If already succeeded or failed, no need to continue.
+      return;
+    }
+
+    LOG.info(project.getName() + ": Installing JxBrowser");
+
+    final File directory = new File(DOWNLOAD_PATH);
+    if (!directory.exists()) {
+      if (!directory.mkdirs()) {
+        LOG.info(project.getName() + ": Unable to create directory for JxBrowser files");
+        status.set(JxBrowserStatus.INSTALLATION_FAILED);
+        return;
+      }
+    }
+
+    // Check for file or loading file
+    final String[] fileNames = {JxBrowserUtils.getPlatformFileName(), JxBrowserUtils.getApiFileName(), JxBrowserUtils.getSwingFileName()};
+    boolean allDownloaded = true;
+    final List<File> files = new ArrayList<>();
+    for (String fileName : fileNames) {
+      final File file = new File(DOWNLOAD_PATH + File.separator + fileName);
+      files.add(file);
+      if (!file.exists()) {
+        allDownloaded = false;
+      }
+    }
+
+    if (allDownloaded) {
+      LOG.info(project.getName() + ": JxBrowser platform file already exists, skipping download");
+      loadClasses(files);
+      return;
+    }
+
+    final File tempLoadingFile = new File(DOWNLOAD_PATH + File.separator + JxBrowserUtils.getLoadingFileName());
+    boolean created = false;
+    try {
+      created = tempLoadingFile.createNewFile();
+    }
+    catch (IOException e) {
+      e.printStackTrace();
+    }
+
+    if (!created) {
+      // This means another IDE already created this file and has started downloading.
+      // Wait for download to finish and then try loading again.
+      LOG.info(project.getName() + ": Waiting for JxBrowser file to download");
```

Comment:
what if the other ide was killed while the file was being downloaded? Now JXBrowser will never work. We need a generally more robust way to do this or we will get a lot of hard to track down bugs. I would suggest we err on the side of a simpler solution than a more efficient one. It is preferable if it wasn't possible for multiple IDEs to be trying to download files to the same locations.

Reply:
Removing this since we're going to do a download per IDE